wagtail/models/__init__.py

```diff
@@ -4381,10 +4381,7 @@ class PageLogEntryQuerySet(LogEntryQuerySet):
     def get_content_type_ids(self):
         # for reporting purposes, pages of all types are combined under a single "Page"
         # object type
-        if self.exists():
```

Comment:
This looks to be switching something else. This isn't guarding against a missing content type, it looks like it's guarding on whether the current queryset has anything in it. I'm not sure where this is used, but it feels like that might be done for a specific reason?

Reply:
Yep - IIRC this is used to populate the "Type" filter in the site history report, by returning a set of the content types that have any log entries against them. In the case of PageLogEntryQuerySet this either contributes `Page` or nothing to the list; this is then combined with the result of LogEntryQuerySet.get_content_type_ids which handles the remaining non-Page models.

Reply:
Sounds like unnecessary 'feature-spread' to me.

Reply:
No, not that. I mean having this method do an `exists()` query so that the behaviour is more convenient for logging. It feels like something the calling code could handle quite easily. Anyhow, it was a throwaway comment. I agree this should be closed.